src/http_ece.rs

```diff
@@ -155,90 +108,44 @@ impl<'a> HttpEce<'a> {
+                base64::encode_config(&encrypted_block.salt, URL_SAFE_NO_PAD)
+            ),
         ));
 
         crypto_headers
     }
 
-    /// The aesgcm encrypted content-encoding, draft 3.
-    pub fn aes_gcm(
-        &self,
-        shared_secret: &'a [u8],
-        as_public_key: &'a [u8],
-        salt_bytes: &'a [u8],
-        payload: &'a mut Vec<u8>,
-    ) -> Result<(), WebPushError> {
-        let mut context = Vec::with_capacity(140);
-
-        context.extend_from_slice("P-256\0".as_bytes());
-        context.push((self.peer_public_key.len() >> 8) as u8);
-        context.push((self.peer_public_key.len() & 0xff) as u8);
-        context.extend_from_slice(self.peer_public_key);
-        context.push((as_public_key.len() >> 8) as u8);
-        context.push((as_public_key.len() & 0xff) as u8);
-        context.extend_from_slice(as_public_key);
-
-        let client_auth_secret = hkdf::Salt::new(hkdf::HKDF_SHA256, &self.peer_secret);
-        let salt = hkdf::Salt::new(hkdf::HKDF_SHA256, salt_bytes);
-
-        let EceKey(prk) = client_auth_secret
-            .extract(shared_secret)
-            .expand(&[&"Content-Encoding: auth\0".as_bytes()], EceKey(32))
-            .unwrap()
-            .into();
-
-        let mut cek_info = Vec::with_capacity(165);
-        cek_info.extend_from_slice("Content-Encoding: aesgcm\0".as_bytes());
-        cek_info.extend_from_slice(&context);
-
-        let EceKey(content_encryption_key) = salt
-            .extract(&prk)
-            .expand(&[&cek_info], EceKey(16))
-            .unwrap()
-            .into();
-
-        let mut nonce_info = Vec::with_capacity(164);
-        nonce_info.extend_from_slice("Content-Encoding: nonce\0".as_bytes());
-        nonce_info.extend_from_slice(&context);
-
-        let EceKey(nonce_bytes) = salt
-            .extract(&prk)
-            .expand(&[&nonce_info], EceKey(12))
-            .unwrap()
-            .into();
-
-        let mut nonce = EceNonce::default();
-        nonce.fill(nonce_bytes);
-
-        let unbound_key = aead::UnboundKey::new(&aead::AES_128_GCM, &content_encryption_key)?;
-        let mut sealing_key = aead::SealingKey::new(unbound_key, nonce);
-
-        sealing_key.seal_in_place_append_tag(aead::Aad::empty(), payload)?;
-
-        Ok(())
-    }
-}
-
-fn front_pad(payload: &[u8], output: &mut [u8]) {
-    let payload_len = payload.len();
-    let max_payload = output.len() - 2;
-    let padding_size = max_payload - payload.len();
-
-    output[0] = (padding_size >> 8) as u8;
-    output[1] = (padding_size & 0xff) as u8;
-
-    for i in 0..payload_len {
-        output[padding_size + i + 2] = payload[i];
+    pub fn generate_headers_aes128gcm(&self) -> Vec<(&'static str, String)> {
+        let mut headers = Vec::new();
+        if let Some(signature) = &self.vapid_signature {
+            headers.push((
+                "Authorization",
+                format!("vapid t={}, k={}", signature.auth_t, signature.auth_k),
```

Comment:
aren't these in the vapid generation already?

Reply:
I'm not sure what you mean... Vapid needs this header (public key + JWT) and it's different on AESGCM (which uses `dh`/`ecdsa` keys in the `Crypto-Key` field, and only the token in `Authorization` with a `WebPush` prefix). The final RFC puts the dh part back in the Authorization field, under the name `k` with a `vapid` prefix, and there is no `Crypto-Key` field. 

Anyway what I mean is the Authorization field is implementation-specific so it's necessarily generated separately, per-implementation, if that's what you meant. This is the part of the code where it was added to the crypto-headers in your previous version (but only the AESGCM version)

Reply:
Yep. I'm not that much into ece encryption anymore, so I trust your research on it... :)

Reply:
I'll do this little clean-up, with the current `Into` implementation looking like it's not AESGCM-specific when it actually is ;)